docs/avro.rst

```diff
@@ -1,98 +1,47 @@
-of type ``string`` and ``int``, respectively.
-
-.. sourcecode:: json
-
-    {"namespace": "example.avro",
-     "type": "record",
-     "name": "user",
-     "fields": [
-         {"name": "name", "type": "string"},
-         {"name": "favorite_number",  "type": "int"}
-     ]
-    }
-
-You can then use this Avro schema, for example, to serialize a Java object (POJO) into bytes, and deserialize these
-bytes back into the Java object.
-
-One of the interesting things about Avro is that it not only requires
-a schema during data serialization, but also during data
-deserialization. Because the schema is provided at decoding time,
-metadata such as the field names don't have to be explicitly encoded
-in the data. This makes the binary encoding of Avro data very compact.
+.. _schema_evolution_and_compatibility:
 
+Schema Evolution and Compatibility
+==================================
 
 Schema Evolution
-^^^^^^^^^^^^^^^^
+----------------
 
 An important aspect of data management is **schema evolution**.  After the initial schema is defined, applications may
 need to evolve it over time. When this happens, it's critical for the downstream consumers to be able to handle data
 encoded with both the old and the new schema seamlessly. This is an area that tends to be overlooked in practice until
 you run into your first production issues.  Without thinking through data management and schema evolution carefully,
 people often pay a much higher cost later on.
 
-There are three common patterns of schema evolution:
+Note: when using Avro, one of the most important things is to manage its schemas and reason about how those
+schemas should evolve. :ref:`Confluent Schema Registry <schemaregistry_intro>` is built for exactly that purpose.
+You can find out the details on how to use it to store Avro schemas and enforce certain compatibility rules during
+schema evolution by looking at the :ref:`schemaregistry_api`.
+
+These are the compatibility types:
+
+.. include:: includes/compatibility_list.rst
 
-1. :ref:`backward compatibility <avro-backward_compatibility>`
-2. :ref:`forward compatibility <avro-forward_compatibility>`
-3. :ref:`full compatibility <avro-full_compatibility>`
+The compatibility types can be grouped into the following four common patterns of schema evolution:
 
+1. :ref:`BACKWARD compatibility <avro-backward_compatibility>`
+2. :ref:`FORWARD compatibility <avro-forward_compatibility>`
+3. :ref:`FULL compatibility <avro-full_compatibility>`
+4. :ref:`no compatibility checking <avro-none_compatibility>`
+
+
+Compatibility Types
+-------------------
 
 .. _avro-backward_compatibility:
 
 Backward Compatibility
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Backward compatibility means that data encoded with an older schema can be read with a newer schema.
+``BACKWARD`` compatibility means that data encoded with an older schema can be read with a newer schema.
```

Comment:
How about this: BACKWARD compatible schema means that version X of the schema can be used to decode data encoded with version X and X-1 of the schema, but not necessarily version X-2 and before.

Reply:
Good idea.  I've adapted this suggestion